AI_NoObjectGeneratedError console error during autocomplete

Seems there is some console error messages during autocomplete, I'm busy investigating...
```
[2025-12-10T06:21:36.781Z] [DEBUG] Firing inline completion request for model gpt-4o at position 123:19
[2025-12-10T06:21:41.079Z] [ERROR] Error in getInlineCompletionItems: No object generated: response did not match schema.
AI_NoObjectGeneratedError: No object generated: response did not match schema.
    at parseAndValidateObjectResult (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/node_modules/ai/dist/index.js:6836:11)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at parseAndValidateObjectResultWithRepair (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/node_modules/ai/dist/index.js:6849:12)
    at fn (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/node_modules/ai/dist/index.js:7186:25)
    at /home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/node_modules/ai/dist/index.js:1644:22
    at generateObject (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/node_modules/ai/dist/index.js:7045:12)
    at OpenAICompatibleProviderClient.getInlineCompleteResponse (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/out/ai/providerClient.js:149:28)
    at getInlineCompletionItems (/home/nkukard/.vscode/extensions/mcowger.generic-copilot-0.13.4/out/autocomplete/fimProvider.js:160:30)
    at Na.provideInlineCompletions (file:///opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:142:155567)
```

Using GPT-4o model via OpenAI via latest LiteLLM.

--- a/src/ai/providerClient.ts
+++ b/src/ai/providerClient.ts
@@ -12,7 +12,7 @@ import { updateContextStatusBar } from "../statusBar";
 import { z } from "zod";
 import * as vscode from "vscode";
 
-import { generateObject, streamText } from "ai";
+import { generateText, streamText } from "ai";
 import { ModelItem, ProviderConfig, VercelType } from "../types";
 import { LM2VercelMessage, LM2VercelTool, normalizeToolInputs } from "./utils/conversion";
 import { ModelMessage, LanguageModel, Provider } from "ai";
@@ -170,20 +170,13 @@ export abstract class ProviderClient {
 	): Promise<string> {
 		const languageModel = this.getLanguageModel(config.slug);
 
-		// The generateObject call
-		const { object } = await generateObject({
+		const result = await generateText({
 			model: languageModel,
-			// Define the structure of the output
-			schema: z.object({
-				completion: z.string().describe(completionDescription),
-			}),
-			// Set the behavior of the model
 			system: completionSystemInstruction,
-
 			// Construct the prompt using the inputs
 			prompt: generateCompletionPromptInstruction(fileName, languageId, prefix, suffix),
 		});
-		return object.completion;
+		return result.text;
 	}
 
 	/**

--- a/src/autocomplete/constants.ts
+++ b/src/autocomplete/constants.ts
@@ -54,7 +54,7 @@ export const FILENAME_SENSITIVE_KEYWORDS = [
 	"wp-config",
 ];
 
-export const completionSystemInstruction = `You are a specialized code completion engine. Your goal is to generate the missing code that bridges the gap between the provided prefix and suffix contexts.
+export const completionSystemInstruction = `You are a specialized code completion engine. Your goal is to generate the missing code that bridges the gap between the provided prefix and suffix contexts.  You must not provide any additional explanations or commentary—only the exact code snippet that fits perfectly between the prefix and suffix.  Limit your suggestions to code only, avoiding any prose or extraneous information.  Ensure that the generated code is syntactically correct and contextually relevant to seamlessly integrate with the surrounding code.  Limit your response to fewer than 5 lines of code.
 `;
 
 export const completionDescription = "The code snippet that fits exactly between the prefix and suffix."

--- a/src/autocomplete/fimProvider.ts
+++ b/src/autocomplete/fimProvider.ts
@@ -81,8 +81,19 @@ function generateCompletionItems(
 	insertText: string,
 	position: vscode.Position
 ): vscode.InlineCompletionItem[] {
-	if (!insertText || !insertText?.trim() || insertText?.trim().length < 9) {
-		logger.debug(`Skip short suggestion: ${insertText}`);
+	switch (insertText) {
+		case null:
+			logger.debug("No insert text provided: null");
+			return [];
+		case undefined:
+			logger.debug("No insert text provided: undefined");
+			return [];
+		case "":
+			logger.debug("No insert text provided: empty string");
+			return [];
+	}
+	if (insertText.trim().length < 9) {
+		logger.debug(`Insert text too short (${insertText.trim().length} chars): "${insertText.trim()}"`);
 		return [];
 	}
 

